Framework/WorkflowAlgorithms/src/QENSFitSequential.cpp

```diff
@@ -0,0 +1,694 @@
+
+  auto outputWs = performFit(inputString, outputBaseName);
+  auto resultWs = processIndirectFitParameters(outputWs);
+  auto groupWs = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(
+      outputBaseName + "_Workspaces");
+  AnalysisDataService::Instance().addOrReplace(
+      getPropertyValue("OutputWorkspace"), resultWs);
+
+  if (containsMultipleData(workspaces))
+    renameWorkspaces(groupWs, spectra, inputWorkspaces);
+  else
+    renameWorkspaces(groupWs, spectra);
+  copyLogs(resultWs, workspaces);
+
+  const bool doExtractMembers = getProperty("ExtractMembers");
+  if (doExtractMembers)
+    extractMembers(groupWs, workspaces, outputBaseName + "_Members");
+
+  deleteTemporaryWorkspaces(outputBaseName);
+
+  addAdditionalLogs(resultWs);
+  copyLogs(resultWs, groupWs);
+
+  setProperty("OutputWorkspace", resultWs);
+  setProperty("OutputParameterWorkspace", outputWs);
+  setProperty("OutputWorkspaceGroup", groupWs);
+}
+
+std::map<std::string, std::string>
+QENSFitSequential::getAdditionalLogStrings() const {
+  bool convolve = getProperty("ConvolveMembers");
+  auto fitProgram = name();
+  fitProgram = fitProgram.substr(0, fitProgram.rfind("Sequential"));
+
+  auto logs = std::map<std::string, std::string>();
+  logs["sample_filename"] = getPropertyValue("InputWorkspace");
+  logs["convolve_members"] = convolve ? "true" : "false";
+  logs["fit_program"] = fitProgram;
+  logs["fit_mode"] = "Sequential";
+  return logs;
+}
+
+std::map<std::string, std::string>
+QENSFitSequential::getAdditionalLogNumbers() const {
+  std::map<std::string, std::string> logs;
+  logs["start_x"] = getPropertyValue("StartX");
+  logs["end_x"] = getPropertyValue("EndX");
+  return logs;
+}
+
+void QENSFitSequential::addAdditionalLogs(
+    MatrixWorkspace_sptr resultWorkspace) {
+  auto logAdder = createChildAlgorithm("AddSampleLog", -1.0, -1.0, false);
+  logAdder->setProperty("Workspace", resultWorkspace);
+
+  Progress logAdderProg(this, 0.99, 1.00, 6);
+  logAdder->setProperty("LogType", "String");
+  for (const auto log : getAdditionalLogStrings()) {
+    logAdder->setProperty("LogName", log.first);
+    logAdder->setProperty("LogText", log.second);
+    logAdder->executeAsChildAlg();
+    logAdderProg.report("Add text logs");
+  }
+
+  logAdder->setProperty("LogType", "Number");
+  for (const auto log : getAdditionalLogNumbers()) {
+    logAdder->setProperty("LogName", log.first);
+    logAdder->setProperty("LogText", log.second);
+    logAdder->executeAsChildAlg();
+    logAdderProg.report("Add number logs");
+  }
+}
+
+std::string QENSFitSequential::getOutputBaseName() const {
+  const auto base = getPropertyValue("OutputWorkspace");
+  auto position = base.rfind("_Result");
+  if (position != std::string::npos)
+    return base.substr(0, position);
+  return base;
+}
+
+bool QENSFitSequential::throwIfElasticQConversionFails() const { return false; }
+
+bool QENSFitSequential::isFitParameter(const std::string &) const {
+  return true;
```

Comment:
Quite bold answer. Should there be some test here?

Reply:
If there are no restrictions by an extending class, all parameters are added to the OutputWorkspace. This is the default behaviour (overrided by ConvolutionFitSequential and IqtFitSequential).